src/medium_test.cpp

```diff
@@ -336,25 +336,28 @@ void test_collected_metrics_content(PQOSMock *p_mock, pqos_cap *p_cap, unsigned
     // Start collecting all metrics
     rdt::Collector *rdt = new rdt::Collector(p_mock);
     auto metrics = fixture_metrics(num_cores);
-    EXPECT_NO_THROW(rdt->collect_metrics(metrics));
+    auto mts = rdt->collect_metrics(metrics);
 
     // Fill expected values map with per-core metric values
     for (int cpu_id = 0; cpu_id < num_cores; cpu_id++) {
         emplace_per_core_metrics(&expected_values, llc_size, cpu_id);
     }
 
     // Checking if collected metrics values equal expected ones
-    for (int i = 0; i < metrics.size(); i++)
+    for (int i = 0; i < mts.size(); i++)
     {
-        auto metric_ns = extract_ns(metrics[i]);
+        std::string metric_ns = "/";
+        metric_ns += mts[i].ns().get_string();
```

Comment:
Leading namespace by separator should happen in get_string(), not outside. I submitted an issue to address this change in snap-plugin-lib-cpp - https://github.com/intelsdi-x/snap-plugin-lib-cpp/issues/66. Until it's not resolved in plugin-lib side, it might stay as it is in collector tests.